versioned/persist/store/src/main/java/org/projectnessie/versioned/persist/store/GenericContentTypeSupplier.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.persist.store;
+
+import com.google.protobuf.ByteString;
+import org.projectnessie.versioned.StoreWorker;
+import org.projectnessie.versioned.persist.adapter.ContentType;
+import org.projectnessie.versioned.persist.adapter.ContentTypeSupplier;
+
+public class GenericContentTypeSupplier<CONTENT, METADATA, CONTENT_TYPE extends Enum<CONTENT_TYPE>>
+    implements ContentTypeSupplier {
+
+  private final StoreWorker<CONTENT, METADATA, CONTENT_TYPE> storeWorker;
+
+  public GenericContentTypeSupplier(StoreWorker<CONTENT, METADATA, CONTENT_TYPE> storeWorker) {
+    this.storeWorker = storeWorker;
+  }
+
+  @Override
+  public ContentType getContentType(ByteString type) {
+    CONTENT_TYPE typeEnum = storeWorker.getType(type);
+    if (typeEnum.name().equals("ICEBERG_TABLE")) {
```

Comment:
Could we reduce the scope of the new information that we need to expose for namespace support so that the code could be generic across different content types? For example, namespaces are generic across different `StoreWorker` types... Would it make sense to only delineate contents vs. namespaces without exposing specific content types?